city/crud.py

```diff
@@ -0,0 +1,37 @@
+from fastapi import HTTPException
+from sqlalchemy import select
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from city import schemas, models
+
+
+async def get_all_cities(db: AsyncSession):
+    query = select(models.City)
+    result = await db.execute(query)
+    cities_list = result.scalars().all()
+    return cities_list
+
+
+async def create_city(db: AsyncSession, city: schemas.CityBase):
+    new_city = models.City(name=city.name, additional_info=city.additional_info)
+    db.add(new_city)
+    await db.commit()
+    await db.refresh(new_city)
+    return new_city
+
+
+async def get_city(db: AsyncSession, city_id: int):
+    try:
+        await db.get(models.City, city_id)
```

Comment:
The `await db.get(models.City, city_id)` should return the city object. You need to assign it to a variable and return it. Otherwise, the `delete_city` function will not work as expected since it relies on the city object being returned.